modules/EnergyManager/EnergyManager.cpp

```diff
@@ -37,6 +37,7 @@ void EnergyManager::ready() {
 
 void EnergyManager::shutdown() {
     invoke_shutdown(*p_main);
+    this->optimizer_thread.stop();
```

Comment:
I'm not sure how the `utils/thread.hpp` should be used, but `stop()` is also called in its destructor, so is it really needed here?